iphone/Classes/KrollContext.m

```diff
@@ -1560,6 +1560,82 @@ - (void)dealloc
 
 #if defined(USE_JSCORE_FRAMEWORK) && !defined(TI_USE_KROLL_THREAD)
 
+@class KrollTimerManager;
+
+/**
+ * Object acting as the target that receives a message when a timer fires.
+ */
+@interface KrollTimerTarget : NSObject
+
+/**
+ * Identifier of the timer this object is a target for. Required to invalidate the timer.
+ */
+@property (strong, nonatomic) NSNumber *timerId;
+
+/**
+ * The JS function to call when the timer fires.
+ */
+@property (strong, nonatomic) JSValue *callback;
+
+/**
+ * Additional arugments to pass to the callback function
+ */
+@property (strong, nonatomic) NSArray<JSValue *> *arguments;
+
+/**
+ * Reference to the timer manager to be able to invalidate one-off timers
+ */
+@property (assign, nonatomic) KrollTimerManager *timerManager;
+
+- (instancetype)initWithCallback:(id)callback arguments:(NSArray<NSValue *> *)arguments timerId:(NSNumber *)timerId timerManager:(KrollTimerManager *)timerManager;
+
+/**
+ * The method that will be triggered when a timer fires.
+ */
+- (void)timerFired:(NSTimer *)timer;
+
+@end
+
+@implementation KrollTimerTarget
+
+- (instancetype)initWithCallback:(id)callback arguments:(NSArray<NSValue *> *)arguments timerId:(NSNumber *)timerId timerManager:(KrollTimerManager *)timerManager
+{
+  self = [super init];
+  if (!self) {
+    return nil;
+  }
+
+  self.callback = callback;
+  self.arguments = arguments;
+  self.timerId = timerId;
+  self.timerManager = timerManager;
+
+  return self;
+}
+
+- (void)dealloc
+{
+  [_callback release];
+  _callback = nil;
+  [_arguments release];
+  _arguments = nil;
+  [_timerId release];
+  _timerId = nil;
+  _timerManager = nil;
+
+  [super dealloc];
+}
+
+- (void)timerFired:(NSTimer *)timer
+{
+  [self.callback callWithArguments:self.arguments];
```

Comment:
one thing that both of us forgot about here was handling exceptions. I suppose if an error/exception gets thrown, the API may just hold onto it in the context and continue on? Should we be registering our exception handler to deal with them? I had just added that locally because I wanted to see what would happen if a timer threw an Error (and it appears to just squash it and move on right now).